main.go

```diff
@@ -136,5 +139,10 @@ func main() {
 		defer undo()
 	}
 
+	// If GOMEMLIMIT is not set, adjust GOMEMLIMIT if running under linux/cgroups quotas.
+	memlimit.SetGoMemLimitWithOpts(memlimit.WithRatio(0.8), memlimit.WithLogger(slog.Default()))
```

Comment:
Probably do not want to use slog here, but I was not sure how to plug it to the server logger, since memlimit.WithLogger expects slog logger.

Reply:
I think we would need to make the server logger implement the `Handler` interface from slog, then here call `slog.New(s.Logger())`